scripts/simulations/simulation_for_1d.py

```diff
@@ -102,21 +102,54 @@
                                  CTCF_right_positions, 
                                  **paramdict)
 
-with h5py.File(folder+"/LEFPositions.h5", mode='w') as myfile:
+with h5py.File(folder+"/LEFPositions.h5", mode = 'w') as myfile:
+    # creating data set for LEF positions
     dset = myfile.create_dataset("positions", 
-                                 shape=(trajectory_length, LEFNum, 2), #edited
-                                 dtype=np.int32, 
-                                 compression="gzip")
-    
+                                 shape = (trajectory_length, LEFNum, 2), #edited
+                                 dtype = np.int32, 
+                                 compression = "gzip")
+
+    # creating data sets for boundary elements possible sites
+    dset_possible_right = myfile.create_dataset("possible_CTCF_right",
+                                                 shape = (len(ctcfrightlist)), compression = "gzip")
+    dset_possible_right[:] = ctcfrightlist[:]
+    dset_possible_left = myfile.create_dataset("possible_CTCF_left",
+                                                shape = len(ctcfleftlist), compression="gzip")
+    dset_possible_left[:]=ctcfleftlist[:]
```

Comment:
I think the data can be passed directluy during the create_dataset
https://docs.h5py.org/en/stable/high/dataset.html
```python
dset = f.create_dataset("init", data=arr)
```

Reply:
I think I needed to determine the shape and compression before creating dataset in order to avoid memory errors.